app/src/main/java/com/anysoftkeyboard/keyboards/views/PointerTracker.java

```diff
@@ -528,7 +527,7 @@ CharSequence getPreviewText(Key key) {
                     ? anyKey.label.toString().toUpperCase(Locale.getDefault())
                     : anyKey.label;
         } else {
-            int multiTapCode = getMultiTapCode(key);
+            int multiTapCode = key.getMultiTapCode(mTapCount);
             // The following line became necessary when we stopped casting multiTapCode to char
             if (multiTapCode < 32) {
```

Comment:
I don't recall what was the assumption here. Probably `printable code`?

Reply:
I'm not sure I understand your question.

`getMultiTapCode` naturally knew `mTapCount` when it was living at `PointerTracker.java`. It lost access to that variable when I moved it to `Keyboard.java`, so I'm passing it as an argument now.

I don't remember why did I make that method non-static. It was probably not strictly necessary, but it seemed more logical when put next to the other functions at `Keyboard.java`.

(If you're talking about the `if (multiTapCode < 32)` condition, it's because when it was a `char`, it could never be negative, so the line creating a String from that codepoint would never crash.)